routes/user.js

```diff
@@ -68,6 +68,29 @@ router.get('/oauth2/google/callback', passport.authenticate('google'), (req, res
     }
 });
 
+//oauth login with github
+router.get('/oauth2/github', passport.authenticate('github'));
+router.get('/oauth2/github/callback', passport.authenticate('github'), (req, res, next) => {
```

Comment:
Do you need to pass a list of OAuth scopes here? If you use the access token to configure the Github device, you need the scopes to be the same: user, public_repo, repo, repo:status, gist, notifications